### PR TITLE
Fix issues in SPAM logic when finding file binaries

### PR DESCRIPTION
**Pull request by kenlautner**
## Description

Fixes in SPAM logic to allow it to find the relevant spam binaries in the FDF files and to also not report a failure incorrectly.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested on Physical system

## Integration Instructions

N/A

### CODE REVIEW

File: SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c [hunk 1]
```diff
@@ -273,8 +275,15 @@ DiscoverSmiEntryInFvHobs (
...
-            mMmiEntryBaseAddress  = (EFI_PHYSICAL_ADDRESS)(UINTN)FileHeader;
-            mMmiEntrySize         = 0;
+            Status = FfsFindSectionData (EFI_SECTION_RAW, FileHeader, &RawMmiEntryFileData, &mMmiEntrySize);
+            if (!EFI_ERROR (Status)) {
+              mMmiEntryBaseAddress  = (EFI_PHYSICAL_ADDRESS)(UINTN)RawMmiEntryFileData;
+            } else {
+              DEBUG ((DEBUG_ERROR, "[%a]   Failed to load MmiEntry [%g] in FV at 0x%p of %x bytes - %r.\n", __FUNCTION__, &gMmiEntrySpamFileGuid, FileHeader, FileHeader->Size, Status));
+              break;
+            }
+            //mMmiEntryBaseAddress  = (EFI_PHYSICAL_ADDRESS)(UINTN)FileHeader;
```
Comment (kuqin12) [inline SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c:285]: Can we remove the comment lines?

File: SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c [hunk 2]
```diff
@@ -329,6 +340,7 @@ DiscoverSmiEntryInFvHobs (
         }
       } while (!EFI_ERROR (Status));
       Hob.Raw = GetNextHob (EFI_HOB_TYPE_FV, GET_NEXT_HOB (Hob));
+      Status = EFI_SUCCESS;
     }
   } while (Hob.Raw != NULL);
```
Comment (kuqin12) [inline SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c:346]: Also, this line below, can we please check the found status for all 3 entities?

**Review by kuqin12 — APPROVED** [review]

File: SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c [hunk 3]
```diff
@@ -291,9 +300,11 @@ DiscoverSmiEntryInFvHobs (
               Status = EFI_ALREADY_STARTED;
               break;
             }
+            Status = FfsFindSectionData (EFI_SECTION_RAW, FileHeader, &RawBinFileData, &SpamBinSize);
```
Comment (kuqin12) [inline SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c:303]: Check status?

**Review by kuqin12 — APPROVED** [review]

File: SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c [hunk 4]
```diff
@@ -329,6 +340,7 @@ DiscoverSmiEntryInFvHobs (
         }
       } while (!EFI_ERROR (Status));
       Hob.Raw = GetNextHob (EFI_HOB_TYPE_FV, GET_NEXT_HOB (Hob));
+      Status = EFI_SUCCESS;
```
Comment (kuqin12) [inline SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c:343]: Do you want to add this right before the `Done:` symbol?

**Review by kuqin12 — APPROVED** [review]

File: SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c (hunk 4 above)
Comment (kenlautner) [reply SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c]: Moved this line to be in the passing condition (found all three binaries).

File: SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c (hunk 1 above)
Comment (kenlautner) [reply SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c]: Removed.

File: SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c (hunk 2 above)
Comment (kenlautner) [reply SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c]: See other comment.

File: SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c (hunk 3 above)
Comment (kenlautner) [reply SpamPkg/Library/SmmCpuFeaturesLib/SmmStm.c:301]: Added a new status check.

**Pull request merged**